### PR TITLE
[CMD] DATE: Simply use short date without weekday

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-13848](https://jira.reactos.org/browse/CORE-13848)

## Proposed changes

- Simply use the result of `GetDateFormat` function in `%DATE%` environment variable and the `DATE` command.

## TODO

- [x] Do tests.

## Comparison

Win2k3:
intl-win2k3

ReactOS (BEFORE):
intl-before

ReactOS (AFTER):
intl-after

### CODE REVIEW

File: base/shell/cmd/locale.c
Comment (HBelusca) [inline base/shell/cmd/locale.c:64]:
```suggestion
    GetDateFormat(LOCALE_USER_DEFAULT, DATE_SHORTDATE, &t, NULL, szDate, ARRAYSIZE(szDate));
```

**Review by HBelusca — APPROVED** [review]
OK, please use the explicit DATE_SHORTDATE flag (even if this is the default if none is specified, it's better to be explicit in this case)

**Review by JoachimHenze — APPROVED** [review]
Approved. Ship it. This is trivial enough to not wait for a 3rd reviewer before the merge. Just wait for the bots.

**Pull request merged**
